paper.md

```diff
@@ -2,11 +2,77 @@
+Since, these container runtimes are available for almost every well known operating systems and architectures, experiments can be reproduced easily using containerized workflows in almost any environment [@stubbs2016endofday] [@zheng_integrating_2015].
+
+<!-- what problem is remaining and what are the contributions -->
+
+Although, there are different container engines available, switching between them is hard as they have different API's and image formats and also due to the absence of tools that allow running containerized workflows in an engine agnostic way.
+It has also been found that as scientific workflows become increasingly complex, continuous validation of the workflows which is critical to ensuring good reproducibility, becomes difficult [@deelman2018future] [@cohen2017scientific].
+Currently, the different container based workflow engines that are available involves engines that are inherently container-native but assume the presence of a fully provisioned Kubernetes [@kubernetes_google] cluster at their disposal for workflow execution. 
+Therefore, this category of workflow engines could also be termed as cloud-native [@balalaie2016microservices]. 
+Some popular examples of this type of workflow engines are Argo [@argocommunity_argoproj_2019], Pachyderm [@novella_containerbased_2018] and Brigade [@brigade]. 
+The presence of a Kubernetes cluster or a cloud computing environment should not be a hardcore requirement for reproducing any experiment in a container-native manner, since it is often costly [@rodriguez2020container] to get access to one and this inturn makes reproducibility complex. 
+Workflow engines should provide flexibility for their users to run workflows in a wide range of computing environments like a single-node local machine, a HPC environment [@yang2005high], a Kubernetes cluster or any other computing environment of their choice. 
+
+Popper [@systemslabpopper] is a light-weight workflow execution engine that follows a container-native strategy for building reproducible workflows from archived experimental artifacts. 
+This paper makes the following contributions:
+
+1. A workflow engine that allows running workflows inside containers on different computing environemnts like a local machine, a cloud computing environment or a HPC environment.
+
+2. The design and architecture of a container-native workflow engine that abstracts resource managers and container engines giving users the ability to focus only on Dockerfiles i.e. software dependencies and workflow logic i.e. correct order of execution, and ignore the runtime specific details.
+
+3. A design that provides built-in support for continuous validation and portability of workflows which empowers researchers to develop workflows once and run interchangeably between CI services like Travis, Jenkins, etc. and the local machine without any modifications. 
```

Comment:
I think 1-3 is the same contribution. So it's:
1. the design/architecture
2. the implementation and discussion
3. case studies
4. comparison

depending of how detailed is the comparison, 4 might end up not being a contribution. All papers have a related work section, so it's kind of a requirement to compare your solution to the existing state-of-the-art ones